spec/requests/providers/link_application/make_links_controller_spec.rb

```diff
@@ -0,0 +1,111 @@
+  let(:legal_aid_application) { create(:legal_aid_application, :with_applicant) }
+  let(:provider) { legal_aid_application.provider }
+
+  describe "GET /providers/applications/:legal_aid_application_id/link_application/make_link" do
+    subject(:get_request) { get providers_legal_aid_application_link_application_make_link_path(legal_aid_application) }
+
+    context "when the provider is not authenticated" do
+      before { get_request }
+
+      it_behaves_like "a provider not authenticated"
+    end
+
+    context "when the provider is authenticated" do
+      before do
+        login_as provider
+        get_request
+      end
+
+      it "shows the link application invitation page" do
+        expect(response).to be_successful
+        expect(unescaped_response_body).to include("Do you want to link this application with another one?")
+      end
+    end
+  end
+
+  describe "PATCH /providers/applications/:legal_aid_application_id/link_application/make_link" do
+    subject(:patch_request) { patch providers_legal_aid_application_link_application_make_link_path(legal_aid_application), params: }
+
+    let(:params) do
+      {
+        linked_application: {
+          link_type_code:,
+        },
+      }
+    end
+    let(:link_type_code) { "FC_LEAD" }
+
+    context "when the provider is not authenticated" do
+      before { patch_request }
+
+      it_behaves_like "a provider not authenticated"
+    end
+
+    context "when the provider is authenticated" do
+      before do
+        login_as provider
+      end
+
+      context "when the Continue button is pressed" do
+        context "when a link type is chosen" do
+          it "redirects to next page" do
+            patch_request
+            # TODO: This will change when ap-4826 is complete
+            expect(response).to redirect_to(providers_legal_aid_application_proceedings_types_path)
+          end
+
+          it "creates a new lead linked application" do
+            expect { patch_request }.to change(LinkedApplication, :count).by(1)
+          end
+
+          it "sets the correct link type" do
+            patch_request
+            expect(legal_aid_application.reload.lead_linked_application.link_type_code).to eq "FC_LEAD"
+          end
+        end
+
+        context "when no is chosen" do
+          let(:link_type_code) { "false" }
+
+          it "redirects to next page" do
+            patch_request
+            expect(response).to redirect_to(providers_legal_aid_application_proceedings_types_path)
+          end
+
+          it "does not create a new lead linked application" do
+            expect { patch_request }.not_to change(LinkedApplication, :count)
+          end
+        end
+
+        context "with no invalid postcode" do
+          let(:link_type_code) { nil }
+
+          it "re-renders the form with the validation errors" do
+            patch_request
+            expect(unescaped_response_body).to include("There is a problem")
```

Comment:
Should there be a 'proper' error message?  `There is a problem` seems a bit//generic!

Reply:
👍 have updated to test for actual error message displayed